cm.json/src/main/java/org/apache/felix/cm/json/io/impl/JsonSupport.java

```diff
@@ -268,84 +270,109 @@ public int read(char[] cbuf, int off, int len) throws IOException {
+                                oldChar = c;
+                                continue;
                             }
                             
-                            // Preserve newline characters inside multiline comments
-                            if (insideComment && !insideLineComment && c == '\n') {
-                                currentLine.append(c);
+                            // Skip over characters inside single-line comments
+                            if (insideLineComment) {
+                                oldChar = c;
                                 continue;
                             }
 
-                            // Skip characters inside multiline comments
-                            if (insideComment && c == '*' && i < off + charsRead - 1 && cbuf[i + 1] == '/') {
+                         // Detect potential end of a multiline comment by detecting a slash that is preceded by a star
+                            if (insideComment && c == '/' && oldChar == '*') {
                                 insideComment = false;
-                                i++; // Skip '/' character
+                                oldChar = c;
                                 continue;
                             }
 
-                            // Skip characters inside single-line comments
-                            if (insideLineComment && c == '\n') {
-                                insideLineComment = false;
+                            // Skip over characters inside multi-line comments but preserve newlines
+                            if (insideComment) {
+                                if(c == '\n') {
+                                    currentLine.append(c);
+                                }
+                                oldChar = c;
+                                continue;
                             }
                         }
-
-                        // Preserve characters outside comments
-                        if (!insideComment && !insideLineComment) {
-                            currentLine.append(c);
-                        }
+                    // Preserve characters outside comments
+                    if (!insideComment && !insideLineComment) {
+                        currentLine.append(c);
                     }
+                    oldChar = c;
+                }
 
-                    filteredContent.append(currentLine.toString());
+                filteredContent.append(currentLine.toString());
 
-                    char[] filteredChars = filteredContent.toString().toCharArray();
-                    int filteredLen = Math.min(filteredChars.length, len);
-                    System.arraycopy(filteredChars, 0, cbuf, off, filteredLen);
-                    return filteredLen;
-                }
-                return charsRead;
+                char[] filteredChars = filteredContent.toString().toCharArray();
+                int filteredLen = Math.min(filteredChars.length, len);
```

Comment:
Wouldn't it be a bug if `filteredChars.length > len`? If so, maybe throwing an exception would be better than smoothing over it.